src/gpu_sysman.c

```diff
@@ -249,6 +249,54 @@ static int gpu_config_free(void) {
   return RET_OK;
 }
 
+/* list GPU metric options that can get disabled at run-time */
+static unsigned int list_gpu_metrics(const gpu_disable_t *disabled) {
+  struct {
+    bool disabled;
+    bool submetric;
+    const char *name;
+  } names[] = {{disabled->engine, false, "Engine"},
+               {disabled->fabric, false, "Fabric port"},
+               {disabled->freq, false, "Frequency"},
+               {disabled->mem, false, "Memory"},
+               {disabled->membw, false, "Memory BW"},
+               {disabled->power, false, "Power"},
+               {disabled->power_ratio, true, "Power ratio"},
+               {disabled->ras, false, "RAS/errors"},
+               {disabled->temp, false, "Temperature"},
+               {disabled->throttle, false, "Throttle time"}};
+
+  unsigned int i;
+
+  if (config.gpuinfo) {
+    unsigned int disabled = 0;
+    INFO("Disabled metrics / submetrics:");
+    for (i = 0; i < STATIC_ARRAY_SIZE(names); i++) {
+      if (names[i].disabled) {
+        INFO("- %s%s", names[i].name, names[i].submetric ? " (submetric)" : "");
+        disabled++;
+      }
+    }
+    if (!disabled) {
+      INFO("- none");
+    }
+    INFO("Enabled metrics:");
+  }
+  unsigned int enabled = 0;
+  for (i = 0; i < STATIC_ARRAY_SIZE(names); i++) {
+    if (!(names[i].disabled || names[i].submetric)) {
+      if (config.gpuinfo) {
+        INFO("- %s", names[i].name);
+      }
+      enabled++;
+    }
+    if (config.gpuinfo && !enabled) {
```

Comment:
Please move this out of the `for` block, otherwise `- none` will be printed for each disabled metric at the beginning of the list. Ironically, it will *not* be printed when there are no metrics, because the loop is never entered.

Reply:
Thanks, I fixed this.

("none" _was_ printed when there were no metrics enabled though, as loop is over whole fixed size array.)